mne/preprocessing/eyetracking/calibration.py

```diff
@@ -0,0 +1,346 @@
+    screen_resolution : array-like
+        The resolution (in pixels) of the screen that the eyetracking data was
+        collected with. For example, ``(1920, 1080)`` for a 1920x1080 resolution
+        display.
+
+    Examples
+    --------
+    Below is an example of data that can be passed to the points parameter:
+    ``>>> data = [(960., 540., 0.23, 950.1,  544.1), (960., 92., 0.38, 967.8, 76. )]``
+    """
+
+    def __init__(
+        self,
+        onset=None,
+        model=None,
+        eye=None,
+        avg_error=None,
+        max_error=None,
+        points=None,
+        screen_size=None,
+        screen_distance=None,
+        screen_resolution=None,
+    ):
+        super().__init__()
+        self["onset"] = onset
+        self["model"] = model
+        self["eye"] = eye
+        self["avg_error"] = avg_error
+        self["max_error"] = max_error
+        if points is not None:
+            self.set_calibration_array(points)
+        else:
+            self["points"] = points
+        self["screen_size"] = screen_size
+        self["screen_distance"] = screen_distance
+        self["screen_resolution"] = screen_resolution
+
+    def __repr__(self):
+        """Return a summary of the Calibration object."""
+        onset = self.get("onset", "N/A")
+        model = self.get("model", "N/A")
+        eye = self.get("eye", "N/A")
+        avg_error = self.get("avg_error", "N/A")
+        max_error = self.get("max_error", "N/A")
+        screen_size = self.get("screen_size", "N/A")
+        screen_distance = self.get("screen_distance", "N/A")
+        screen_resolution = self.get("screen_resolution", "N/A")
+        return (
+            f"Calibration |\n"
+            f"  onset: {onset} seconds\n"
+            f"  model: {model}\n"
+            f"  eye: {eye}\n"
+            f"  average error: {avg_error} degrees\n"
+            f"  max error: {max_error} degrees\n"
+            f"  screen size: {screen_size} meters\n"
+            f"  screen distance: {screen_distance} meters\n"
+            f"  screen resolution: {screen_resolution} pixels\n"
+        )
+
+    def __getattr__(self, name):
+        """Allow dot indexing of dict keys."""
+        if name in self:
+            return self[name]
+        raise AttributeError(
+            f"'{self.__class__.__name__}' object has no attribute '{name}'"
+        )
+
+    def copy(self):
+        """Copy the instance.
+
+        Returns
+        -------
+        info : instance of Calibration
+            The copied Calibration.
+        """
+        return deepcopy(self)
+
+    def set_calibration_array(self, data):
+        """
+        Create a Numpy Array containing data regarding each calibration point.
+
+        This method takes an array-like objects and converts it into a structured numpy
+        array, with field names ``'point_x'``, ``'point_y'``, ``'offset'``,
+        ``'gaze_x'``, and ``'gaze_y'``.
```

Comment:
I would avoid the structured array. They are often a pain to access and set etc. To me it's cleaner just to have `self['points']` and `self['gaze']` each of shape `(N, 2)` and `self['offset']` of shape `(N,)` for example.

And to me this `set_calibration_data` really seems like an implementation detail of the `read_eyelink_calibration` function. I think we should get rid of the `__init__` of this function altogether and just have `read_eyelink_calibration` set the `Calibration(..., points=..., gaze=..., offset=...)` properly in the first place.

Reply:
Yeah.. I liked the way the structured array kept all the info in one place but totally agree that it's a pretty arcane thing. I was struggling to find a similar implementation with `dict` but your suggestion is nice and i'll implement it and remove `set_calibration_array`.